src/coreclr/jit/gtlist.h

```diff
@@ -53,6 +53,8 @@ GTNODE(ARR_LENGTH       , GenTreeArrLen      ,0,(GTK_UNOP|GTK_EXOP))      // arr
 GTNODE(INTRINSIC        , GenTreeIntrinsic   ,0,(GTK_BINOP|GTK_EXOP))     // intrinsics
 
 GTNODE(LOCKADD          , GenTreeOp          ,0,(GTK_BINOP|GTK_NOVALUE))
+GTNODE(XAND             , GenTreeOp          ,0,GTK_BINOP)
+GTNODE(XXOR             , GenTreeOp          ,0,GTK_BINOP)
```

Comment:
Why is it named `XXOR` - without knowing the context I would assume it represents *eXclusive* OR atomic operation. 
In fact, there is `LDEORAL` arm64 instruction that would correspond to such node if we decided to introduce `Interlocked.Xor` operation.

I would suggest renaming the node as `XORR` (`ORR` corresponds to bitwise OR instruction on arm64) unless someone from @dotnet/jit-contrib can come up with a better name.

I wonder if we should put the new nodes under `TARGET_ARM64` - there is no much we can do in terms of optimizing their codegen on `XARCH` (see https://godbolt.org/z/Pqza89). In theory, we could optimize no-return versions but I doubt we would be able to tell if the return valued is not consumed in importer where we intrinsify calls to `Interlocked.And/Or`. @dotnet/jit-contrib Any opinion on this?

Reply:
yeah I wasn't happy with `XXOR` either, XORR sounds good to me 👍

Reply:
@echesakovMSFT addressed.

Regarding XArch I think it makes sense to optimize because these And and Or are used in tasks without fetching the return value. But not as part of this PR of course.